src/router/storeSync/LayerParamConfig.class.js

```diff
@@ -78,26 +138,28 @@ function dispatchLayersFromUrlIntoStore(store, urlParamValue) {
         ) {
             log.debug(`  Add layer ${layer.id} if not present`)
             // checking if it is an external layer first
-            if (layer.id.startsWith('KML|') && layer.id.split('|').length === 3) {
-                const splittedLayerId = layer.id.split('|')
-                const kmlLayer = new KMLLayer(
-                    layer.opacity,
-                    splittedLayerId[1],
-                    null,
-                    layer.customAttributes.adminId
-                )
-
-                promisesForAllDispatch.push(
-                    getKmlMetadata(kmlLayer.fileId, kmlLayer.adminId)
-                        .then((metadata) => {
-                            kmlLayer.metadata = metadata
-                            return store.dispatch('addLayer', kmlLayer)
-                        })
-                        .catch((error) => {
-                            log.error(`Failed to get KML metadata for ${splittedLayerId[1]}`, error)
-                            return store.dispatch('addLayer', kmlLayer)
-                        })
-                )
+            const externalLayer = transformParsedExternalLayerIntoObject(layer)
+            if (externalLayer) {
+                // special case for KML :
+                // Set the kmlIds in the drawing module in order to edit it.
```

Comment:
Incorrect comment, should be
`Get and attached KML metadata from backend, this is needed for the drawing module in order to allow (or not) kml editing`

Reply:
Done